btree/btree.c

```diff
@@ -2681,6 +2685,23 @@ static void m0_btree_put_split_and_find(struct nd *l_alloc , struct nd *l_node,
 	diff = m0_bufvec_cursor_cmp(&cur_1, &cur_2);
 	tgt->s_node = diff < 0 ? l_slot.s_node : r_slot.s_node;
 
+	/**
+	 * Corner case: If given record needs to be inseted at internal left
+	 * node and if the key of given record is greater than key at last index
+	 * of left record, initialised tgt->s_idx explicitly, as node_find will
+	 * not compare key with last indexed key.
+	 */
+	if (node_level(tgt->s_node) > 0 && tgt->s_node == l_slot.s_node) {
+		l_slot.s_idx = node_count(l_slot.s_node);
+		l_slot.s_rec = temp_rec;
+		node_key(&l_slot);
+		m0_bufvec_cursor_init(&cur_2, &r_slot.s_rec.r_key.k_data);
+		diff = m0_bufvec_cursor_cmp(&cur_1, &cur_2);
+		if (diff > 0) {
+			tgt->s_idx = node_count(l_slot.s_node) + 1;
+			return;
+		}
+	}
```

Comment:
We need to compare the given record key with last indexed key from left node as node_find will not consider last indexed key for comparison in case of internal node . But in this case we are moving records from right node to left node, we might want to compare given record key with last indexed key from left node, as if it is greater than last indexed key from left node, we might want to add given record at the end of left node.
example, [10,20,30,40]node keys in internal  left node after moving records form right to left. our key is 50, we want index 4 as result but node_find will give 3 as result as it will not consider 40 for comparison as it is internal node